Applications/src/detect-cardiac-phases.cc

```diff
@@ -0,0 +1,132 @@
+/*
+ * Medical Image Registration ToolKit (MIRTK)
+ *
+ * Copyright 2008-2015 Imperial College London
+ * Copyright 2008-2013 Daniel Rueckert, Julia Schnabel
+ * Copyright 2013-2015 Andreas Schuh
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mirtk/Options.h"
+#include "mirtk/IOConfig.h"
+
+#include "mirtk/BaseImage.h"
+#include "mirtk/GenericImage.h"
+#include "mirtk/GaussianBlurring.h"
+
+using namespace mirtk;
+
+char *cine_name = NULL;
+char *out_ED_name = NULL;
+char *out_ES_name = NULL;
+
+//void usage()
+//{
+//	cerr << "Usage: detect-cardiac-phases [cine] [ED output] [ES output]" << endl;
+//	exit(1);
+//}
+
+void PrintHelp(const char *name)
+{
+  cout << endl;
+  cout << "Usage: " << name << " <cine> <ED output> <ES output>" << endl;
+  cout << endl;
+  cout << "Description:" << endl;
+  cout << "  This program detects ED and ES phases from a cine image sets." << endl;
+  cout << endl;
+  cout << "Arguments:" << endl;
+  cout << "  cine        Input cine image sets." << endl;
+  cout << "  ED output   ED phase image." << endl;
+  cout << "  ES output   ES phase image." << endl;
+  cout << endl;
+  PrintCommonOptions(cout);
+  cout << endl;
+}
+
+int main(int argc, char* argv[] )
+{
+	int ok,i,j,k,t,esphase,frames;
+    short cine_max,cine_min,cinedis;
+	double *similarity,*smoothsimilarity,dif;
+	// Check command line
+	REQUIRES_POSARGS(3);
+	InitializeIOLibrary();
+
+	// Parse source and target images
+	cine_name = POSARG(1);
+	out_ED_name = POSARG(2);
+    out_ES_name = POSARG(3);
+
+    // Create images
+    GreyImage cine;
+    cine.Read(cine_name);
+    GreyImage blured;
+    blured.Initialize(cine.GetImageAttributes());
+
+    GaussianBlurring<GreyPixel> gaussianBlurring(2);
+    gaussianBlurring.Input (&cine);
+    gaussianBlurring.Output(&blured);
+    gaussianBlurring.Run();
+
+    ImageAttributes atr = cine.GetImageAttributes();
+    similarity = new double[atr._t];
```

Comment:
Another relict from old IRTK habits There is no need for a dynamically allocated C-style array here. In case of premature return or exception, memory is not freed properly (though arguably less of an issue in the `main` function, but still not a good practice). The use of `std::vector` is appropriate here. Respectively in MIRTK, it's called `mirtk::Array`, i.e.,

```
Array<double> similarity(atr._t, 0.0);
```

There is also no need to initialize the array below then in a for loop over `frames`.

(also a minor optional note, I usually name variables of image attributes `attr`)